libs/platform/src/lib/form/multi-combobox/commons/base-multi-combobox.ts

```diff
@@ -362,11 +365,18 @@ export abstract class BaseMultiCombobox extends CollectionBaseInput implements A
         @Optional() readonly dialogConfig: DialogConfig,
         protected multiComboboxConfig: MultiComboboxConfig,
         @Optional() @SkipSelf() @Host() formField: FormField,
-        @Optional() @SkipSelf() @Host() formControl: FormFieldControl<any>
+        @Optional() @SkipSelf() @Host() formControl: FormFieldControl<any>,
+        @Inject(MAP_LIMIT) private _mapLimit: number
     ) {
         super(_cd, ngControl, ngForm, formField, formControl);
     }
 
+    ngOnChanges(changes: SimpleChanges): void {
+        if (changes['dataSource']) {
+            this._initializeDataSource(this.dataSource);
```

Comment:
This is still will be different. You have `this._dataSource` changed in `_initializeDataSource` and previously, with getter approach, we were returning `this._datasource` in getter of `this.datasource`. And now `this.datasource` will be returned. Please, just revert this to its oiriginal state.

Reply:
@platon-rov which original state? you mean to remove `ngOnChanges` at all and return to setters and getters for datasource?

Reply:
yea, exactly

Reply:
@platon-rov If we return to setter and getter approach, then we'll face the following issue again: 
<img width="799" alt="image" src="https://user-images.githubusercontent.com/3863691/169517649-3b23b16e-3541-4a97-9ca1-c243fccde25f.png">

Reply:
Okay, i see. Reasonable.

Anyway, we need to partially change this behavior to its previous state. Let's not initialize ds in setter, but we should left those accessors and also set `_dataSource` in `_initializeDataSource` method. 

```js
set dataSource(...
  // do nothing
}
get dataSource(...
  // we wanna persist this one
  return this._dataSource;
}

ngOnChanges(...
  if(changes.dataSource) {
    // in this method we will set this._dataSource as previously
    this._initializeDataSource(changes.dataSource.currentValue);
}
```

Reply:
@platon-rov done, please check